src/main/java/com/google/sps/firestore/FirestoreUtil.java

```diff
@@ -0,0 +1,60 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.PROJECT_ID;
+
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import java.io.IOException;
+import javax.servlet.ServletException;
+
+/** Util methods for Firestore. */
+public class FirestoreUtil {
+
+  private static Firestore db;
+  private FirestoreUtil() {}
+
+  /**
+  * Gets the database instance for this project. Initializes database if it
+  * has not been initialized.
+  * @return Firestore database instance
+  * @throws IOException if database initialization fails.
+  */
+  public static Firestore getDatabase() {
+    if (db == null) {
+      try {
+        db = initializeDatabase();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
```

Comment:
Is this meaningful? What would you actually be able to do if your db fails to initialize, given that all of your operations are basically get or set on the database? This seems like it will just obscure the cause of the failure when something in on of your other classes hits a null

Reply:
Changed this method to throws exception and put the try/catch in the servlet calls. HTTPResponse is set to a fail code if there is an exception in doPost().

Note: tried throwing exception in servlet init() but it causes the servlet to crash if there is an exception.